src/macros.rs

```diff
@@ -212,3 +212,212 @@ macro_rules! doc {
         object
     }};
 }
+
+/// Construct a [`crate::RawBson`] value from a literal.
+///
+/// ```rust
+/// use bson::rawbson;
+///
+/// let value = rawbson!({
+///     "code": 200,
+///     "success": true,
+///     "payload": {
+///       "some": [
+///           "pay",
+///           "loads",
+///       ]
+///     }
+/// });
+/// ```
+#[macro_export]
+macro_rules! rawbson {
+    //////////////////////////////////////////////////////////////////////////
+    // TT muncher for parsing the inside of an array [...]. Produces a
+    // RawArrayBuf containing the elements.
+    //
+    // Must be invoked as: bson!(@array [] $($tt)*)
+    //////////////////////////////////////////////////////////////////////////
+
+    // Finished with trailing comma.
+    (@array [$($elems:expr,)*]) => {
+        <$crate::RawArrayBuf as std::iter::FromIterator::<$crate::RawBson>>::from_iter(vec![$($elems,)*])
```

Comment:
This isn't as efficient as it could be, but it was a lot easier then rewriting this to build up the `RawArrayBuf` like we do for `RawDocumentBuf`. I field RUST-1118 to cover the work for doing this without the `Vec` allocation.